Exercise_1.java

```diff
@@ -1,8 +1,21 @@
 class BinarySearch { 
     // Returns index of x if it is present in arr[l.. r], else return -1 
     int binarySearch(int arr[], int l, int r, int x) 
-    { 
-        //Write your code here
+    {
+        if (r>=l)
+        {
+            int mid = l + (r - l)/2;

```

Comment:
Can we use mid = (l+r)/2; here. Give reason along with your answer.

Reply:
> Can we use mid = (l+r)/2; here. Give reason along with your answer - It can lead to overflow. It fails for large values of the int variables low and high.